packages/frontend/src/Build.spec.ts

```diff
@@ -745,28 +763,52 @@ test('select anaconda-iso and qcow2 and expect validation error to be shown', as
 
   // Get checkbox 'iso-checkbox' and click it.
   const iso = screen.getByLabelText('iso-checkbox');
-  iso.click();
-
-  // Give time to propagate the changes
-  await new Promise(resolve => setTimeout(resolve, 300));
+  await userEvent.click(iso);
 
   // Expect 'alert' to not be there
   expect(screen.queryByRole('alert')).toBeNull();
 
   // Get checkbox 'qcow2-checkbox' and click it.
   const qcow2 = screen.getByLabelText('qcow2-checkbox');
-  qcow2.click();
-
-  // Give time to propagate the changes
-  await new Promise(resolve => setTimeout(resolve, 300));
+  await userEvent.click(qcow2);
 
   // Expect alert to be shown
   const validation = screen.getByRole('alert');
   expect(validation).toBeDefined();
   expect(validation.textContent).toEqual(
     'The Anaconda ISO file format cannot be built simultaneously with other image types.',
   );
+});
+
+test('confirm successful build goes to logs', async () => {
+  vi.mocked(bootcClient.listHistoryInfo).mockResolvedValue(mockHistoryInfo);
+  vi.mocked(bootcClient.listBootcImages).mockResolvedValue(mockBootcImages);
+  vi.mocked(bootcClient.checkPrereqs).mockResolvedValue(undefined);
+  vi.mocked(bootcClient.buildExists).mockResolvedValue(true);
+  vi.mocked(bootcClient.inspectImage).mockResolvedValue(mockImageInspect);
+
+  // mock unique id so that it matches an image in the history ('new' build exists already)
+  vi.mocked(bootcClient.generateUniqueBuildID).mockReturnValue(Promise.resolve(mockHistoryInfo[0].id));
+
+  render(Build);
+
+  // Wait until children length is 2 meaning it's fully rendered / propagated the changes
+  while (screen.getByLabelText('image-select')?.children.length !== 2) {
```

Comment:
Why don't you use vi.waitUntil

Reply:
vi.waitFor

Reply:
Yes, that is better. Was just copying what was there but switched all 9 instances to vi.waitFor(). Seems like there should be a more succinct version but it escapes me atm.